src/pkg/util/fs/mount/mount.go

```diff
@@ -0,0 +1,370 @@
+	{"noatime", 0},
+	{"nodev", syscall.MS_NODEV},
+	{"nodiratime", 0},
+	{"noexec", syscall.MS_NOEXEC},
+	{"noiversion", 0},
+	{"nolazytime", 0},
+	{"nomand", 0},
+	{"norelatime", 0},
+	{"nostrictatime", 0},
+	{"nosuid", syscall.MS_NOSUID},
+	{"rbind", syscall.MS_BIND | syscall.MS_REC},
+	{"relatime", 0},
+	{"remount", syscall.MS_REMOUNT},
+	{"ro", syscall.MS_RDONLY},
+	{"rw", 0},
+	{"silent", syscall.MS_SILENT},
+	{"strictatime", 0},
+	{"suid", 0},
+	{"sync", 0},
+}
+
+type fsContext struct {
+	context bool
+}
+
+var authorizedImage = map[string]fsContext{
+	"ext3":     {true},
+	"squashfs": {true},
+}
+
+var authorizedFS = map[string]fsContext{
+	"overlay": {true},
+	"tmpfs":   {true},
+	"ramfs":   {true},
+	"devpts":  {true},
+	"sysfs":   {false},
+	"proc":    {false},
+	"mqueue":  {false},
+}
+
+// ConvertOptions converts an options string into a pair of mount flags and mount options
+func ConvertOptions(options []string) (uintptr, []string) {
+	var flags uintptr
+	finalOptions := []string{}
+	isFlag := false
+
+	for _, option := range options {
+		optionTrim := strings.TrimSpace(option)
+		for _, flag := range mountFlags {
+			if flag.option == optionTrim {
+				flags |= flag.flag
+				isFlag = true
+				break
+			}
+		}
+		if !isFlag {
+			finalOptions = append(finalOptions, optionTrim)
+		}
+		isFlag = false
+	}
+	return flags, finalOptions
+}
+
+// Points defines and stores a set of mount points
+type Points struct {
+	Context string
+	points  []specs.Mount
+}
+
+func (p *Points) add(source string, dest string, fstype string, flags uintptr, options string) error {
+	var bind = false
+
+	mountOptions := []string{}
+
+	if dest == "" {
+		return fmt.Errorf("mount point must contain a destination")
+	}
+	if !strings.HasPrefix(dest, "/") {
+		return fmt.Errorf("destination must be an absolute path")
+	}
+	for i := len(mountFlags) - 1; i >= 0; i-- {
+		flag := mountFlags[i].flag
+		if flag != 0 && flag == (flags&flag) {
+			if bind && flag&syscall.MS_BIND != 0 {
+				continue
+			}
+			mountOptions = append(mountOptions, mountFlags[i].option)
+			if flag&syscall.MS_BIND != 0 {
+				bind = true
+			}
+		}
+	}
+	setContext := true
+	for _, option := range strings.Split(options, ",") {
+		o := strings.TrimSpace(option)
+		if o != "" {
+			if strings.HasPrefix(o, "context=") {
+				setContext = false
+			}
+			mountOptions = append(mountOptions, o)
+		}
+	}
+	if fstype != "" && setContext {
+		setContext = authorizedFS[fstype].context
+	}
+	if setContext {
+		context := fmt.Sprintf("context=%s", p.Context)
+		mountOptions = append(mountOptions, context)
+	}
+	p.points = append(p.points, specs.Mount{
+		Source:      source,
+		Destination: dest,
+		Type:        fstype,
+		Options:     mountOptions,
+	})
+	return nil
+}
+
+// GetAll returns all registered mount points
+func (p *Points) GetAll() []specs.Mount {
+	return p.points
+}
+
+// Get returns registered mount points with the matched destination
+func (p *Points) Get(dest string) []specs.Mount {
```

Comment:
Does it make more sense to get the point associated with the src rather than dest? Not sure the correct answer here, just a thought! Or maybe provide:

`func GetFromSource(src string)`
`func GetFromDest(dest string)`

So user of package can choose

Reply:
Source can be empty (for remount point) or set to none for FS, dest ensure you get the right mount and unique mount point

Reply:
I see! I think it works, but maybe nice to provide option to search for source as well?